analyses/focal-cn-file-preparation/01-prepare-cn-file.R

```diff
@@ -76,24 +76,34 @@ process_annotate_overlaps <- function(cnv_df,
   cnv_gr <- cnv_df %>%
     GenomicRanges::makeGRangesFromDataFrame(keep.extra.columns = TRUE,
                                             starts.in.df.are.0based = FALSE)
-
-  # Create a data.frame with the overlaps between the CNV file and hg38 genome
-  # annotations
-  overlaps <- IRanges::mergeByOverlaps(cnv_gr, tx_exons)
-
-  # Create a data.frame with selected columns from the `autosome_overlaps`
+  
+  # Find the overlaps between the CNV file and hg38 genome annotations
+  overlaps <- GenomicRanges::findOverlaps(cnv_gr, exons_gr)
+  
+  # Carry over samples to `cnv_gr` object
+  cnv_gr@elementMetadata@listData$org_id <- rep(NA, length(cnv_gr))
+  
+  cnv_gr@elementMetadata@listData$org_id[overlaps@from] <-
+    exons_gr$gene_id[overlaps@to]
+  
+  # Find the overlaps between the CNV file and UCSC cytoband data
+  overlaps_ucsc <- GenomicRanges::findOverlaps(cnv_gr, ucsc_cytoband_gr)
+  
+  # Create a data.frame with selected columns from the `overlaps`
   # object
   annotated_cn <- data.frame(
-    biospecimen_id = overlaps$Kids_First_Biospecimen_ID,
-    status = overlaps$status,
-    copy_number = overlaps$copy_number,
-    ploidy = overlaps$tumor_ploidy,
-    ensembl = unlist(overlaps$gene_id),
+    biospecimen_id = cnv_gr$Kids_First_Biospecimen_ID[overlaps_ucsc@from],
+    status = cnv_gr$status[overlaps_ucsc@from],
+    copy_number = cnv_gr$copy_number[overlaps_ucsc@from],
+    ploidy = cnv_gr$tumor_ploidy[overlaps_ucsc@from],
+    ensembl = cnv_gr$org_id[overlaps_ucsc@from],
+    cytoband = ucsc_cytoband_gr@elementMetadata@listData$cytoband[overlaps_ucsc@to],
     stringsAsFactors = FALSE
   ) %>%
     dplyr::distinct() %>%
```

Comment:
What's this distinct doing?

Reply:
Were you able to remove this distinct or determine what it was doing, @cbethell ?